pandas/core/reshape/reshape.py

```diff
@@ -600,6 +600,24 @@ def stack_multiple(frame, level, dropna=True):
     return result
 
 
+def _stack_multi_column_index(columns):
+    if len(columns.levels) <= 2:
+        return columns.levels[0]._rename(name=columns.names[0])
+    levs = []
+    for lev, level_codes in zip(columns.levels[:-1], columns.codes[:-1]):
+        if -1 in level_codes:
+            lev = np.append(lev, None)
+        levs.append(np.take(lev, level_codes))
+    dedupe_levs = zip(*(key for key, _ in itertools.groupby(zip(*levs))))
```

Comment:
can you comment on what this line is doing?  (id also suggest keeping it in two lines like  in the existing code)

Reply:
The purpose is to remove duplicate tuples in the MultiIndex.

For example, if we inspect the `stack` call on line 108 of `test_stack_unstack.py`, we see the following:

```
>>> columns
MultiIndex([('w', 'j', 'a'),
            ('w', 'j', 'b'),
            ('w', 'k', 'a')],
           names=[None, 'y', 'x'])
>>> columns
MultiIndex([('w', 'j', 'a'),
            ('w', 'j', 'b'),
            ('w', 'k', 'a')],
           names=[None, 'y', 'x'])
>>> levs
[Index(['w', 'w', 'w'], dtype='object'), Index(['j', 'j', 'k'], dtype='object', name='y')]
>>> list(dedupe_levs)
[('w', 'w'), ('j', 'k')]
```

The original code used the deduped tuples directly (`[('w', 'j'), ('w', 'k')]`) while the new code coverts them back into levels that can be used by `from_arrays`.

I've separated the line and added a comment.